lib/onelogin/ruby-saml/response.rb

```diff
@@ -863,8 +863,12 @@ def validate_signature
           valid = false
           expired = false
           idp_certs[:signing].each do |idp_cert|
-            valid = doc.validate_document_with_cert(idp_cert)
+            valid = doc.validate_document_with_cert(idp_cert, true)
```

Comment:
It should be used
```
valid = doc.validate_document_with_cert(idp_cert, @soft)
```
instead
```
 valid = doc.validate_document_with_cert(idp_cert, true)
```

Reply:
we cannot use the `@soft` variable here, as it would raise if any of the certificates is invalid. this validation is required to be soft.

I will adjust the code as per your second comment. and using the soft variable to determine if it should raise after validating all certificates